docs/extending/localtunneldebugger.md

```diff
@@ -0,0 +1,143 @@
+| Kubernetes extension | <ul><li>Display the Debug (Local Tunnel) command in the palette or when interacting with supported resources in the Cluster Explorer</li><li>Allow users to configure which debug provider to use</li></ul>                               |                              |
+
+## Implementing the local tunnel debug provider
+
+A local tunnel debug provider must implement the following interface.  (For documentation purposes
+the interface is written in TypeScript terms but any JavaScript object that provides
+the specified properties and methods will do.)
+
+```javascript
+interface LocalTunnelDebugProvider {
+    readonly id: string;
+    startDebugging(target?: any): void;
+}
+```
+
+### Implementing the metadata
+
+The `id` should be a user-friendly name for your debug provider, such as `Contoso Trampoline Debugger`.  
+The `id` will allow users to configure a default local tunnel debugger to use by setting the `vs-kubernetes.local-tunnel-debug-provider` field under `vs-kubernetes` in settings.json.
+
+### Implementing the debugger
+
+Your provider's implementation of `startDebugging(target?: any)` is responsible for resolving the debug target (such as a pod or service) and starting the debug session. You can use the cluster explorer API to resolve the target (see [commandtargets.md](commandtargets.md) for more info). An example implementation is given below:
+
+```javascript
+const MY_PROVIDER = {
+    id: "Contoso Trampoline Debugger",
+    startDebugging: (target?: any) => startDebugSession(target)
+};
+
+function startDebugSession(target?: any) {
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+    const commandTarget = clusterExplorer.resolveCommandTarget(target);
+
+    if (commandTarget) {
+        if (commandTarget.nodeType === 'resource') {
+            if (commandTarget.resourceKind.manifestKind === 'Pod') {
+
+                // Custom logic for selecting or generating the Local Tunnel debug configuration goes here
+                const configToUse: string | vscode.DebugConfiguration = this.resolveLocalTunnelDebugConfigurationForPod(commandTarget);
+
+                // Start the debug session
+                vscode.debug.startDebugging(folder, configToUse);
+            }
+        }
+    } else {
+        // The user selected the Debug (Local Tunnel) command from the command palette or from
+        // somewhere else that you were displaying it.
+    }
+}
+```
+
+## Registering the debug provider
+
+In order to hook into the Debug (Local Tunnel) option, a local debug provider must be _registered_
+with the Kubernetes extension. This is the responsibility of the VS Code extension that hosts
+the debug provider. To do this, the extension must:
+
+* Activate in response to the `kubernetes.cloudExplorer` view
```

Comment:
I think you indicated Cluster Explorer rather than Cloud Explorer before?

In any case, is this actually required or could an extension register only when the user issues the command?  (Perhaps there are timing issues registering in response to the command - if so we should consider those.)

Reply:
Yes, you're right, thank you for catching that. Regarding the timing, I agree that it shouldn't be a requirement for the extension to activate unless the actual command is getting executed. Perhaps I can make this bullet "suggested" rather than a "must"